src/helpers/buildWalletSections.tsx

```diff
@@ -412,27 +441,46 @@ const withBriefBalanceSection = (
       type: 'PROFILE_NAME_ROW_SPACE_AFTER',
       uid: 'profile-name-space-after',
     },
-    {
-      type: 'PROFILE_BALANCE_ROW',
-      uid: 'profile-balance',
-      value: totalValue,
-    },
-    {
-      type: 'PROFILE_BALANCE_ROW_SPACE_AFTER',
-      uid: 'profile-balance-space-after',
-    },
+    ...(hasTokens
+      ? [
+          {
+            type: 'PROFILE_BALANCE_ROW',
+            uid: 'profile-balance',
+            value: totalValue,
+          },
+          {
+            type: 'PROFILE_BALANCE_ROW_SPACE_AFTER',
+            uid: 'profile-balance-space-after',
+          },
+        ]
+      : []),
     {
       type: 'PROFILE_ACTION_BUTTONS_ROW',
       uid: 'profile-action-buttons',
       value: totalValue,
     },
-    {
-      type: 'PROFILE_ACTION_BUTTONS_ROW_SPACE_AFTER',
-      uid: 'profile-action-buttons-space-after',
-      value: totalValue,
-    },
-    ...(isLoadingAssets ? LOADING_ASSETS_PLACEHOLDER : briefAssets),
+    hasTokens
+      ? {
+          type: 'PROFILE_ACTION_BUTTONS_ROW_SPACE_AFTER',
+          uid: 'profile-action-buttons-space-after',
+          value: totalValue,
+        }
+      : { type: 'BIG_EMPTY_WALLET_SPACER', uid: 'big-empty-wallet-spacer-1' },
   ];
+
+  let content = CONTENT_PLACEHOLDER;
+
+  if (isEmpty) {
+    content = EMPTY_WALLET_CONTENT;
+  } else if (hasNFTsOnly) {
+    content = ONLY_NFTS_CONTENT;
+  } else if (isLoadingAssets) {
+    content = CONTENT_PLACEHOLDER;
+  } else if (hasTokens) {
```

Comment:
wonder if this should just be else so we dont get into any weird states cause I dont trust out data flow wdyt?

Reply:
what content should be attempted to be shown in the else case?